Add customer data warning in troubleshooting (#18625)
DataDog/documentation: content/en/real_user_monitoring/browser/troubleshooting.md

Example:

```javascript

> window.DD_RUM && window.DD_RUM.setGlobalContextProperty('global', {'foo': 'bar'})

Author: `addRumGlobalContext` is deprecated in favour of `setGlobalContextProperty`

| Maximum event size                       | 256 KB       |
| Maximum intake payload size              | 5 MB         |

## Customer data exceeds the recommended 3KiB warning

> The RUM browser SDK allows you to set [global context](https://docs.datadoghq.com/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#global-context), [user information](https://docs.datadoghq.com/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#user-session) and [feature flags](https://docs.datadoghq.com/real_user_monitoring/guide/setup-feature-flag-data-collection/?tab=browser) which are then included with the collected events.

Reviewer: Hi @amortemousque , these should use a relative linking format :)

> To minimize the user bandwidth impact, the RUM browser SDK throttles the data sent to the Datadog intake. However sending large volumes of data can still impact the performance for users on slow internet connections.

Reviewer: ```suggestion
To minimize the user bandwidth impact, the RUM browser SDK throttles the data sent to the Datadog intake. However, sending large volumes of data can still impact the performance for users on slow internet connections.
```

> To prevent impacting the user bandwidth the RUM browser SDK throttles the data sent to the Datadog intake. However, to be able to handle heavy RUM traffic it does not cover lower connectivity (<4g). If a large volume of data is sent on a low-connectivity network, it can slow down the requests made by the host application.

Reviewer 2: <!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏That is not clear to me, what do you mean by "it does not cover lower connectivity"?

Reviewer 3: Agreed. It might be interpreted as we don't support lower connectivities.

Reviewer 4: Is something like this accurate?
"Note: sending large volumes of context data can impact performance for users on slow internet connections."

However, I'm also wondering if we should directly update the console message instead of adding this to our documentation. What do you think?

Author: I've updated the documentation according to the RFC.

> Therefore, it is recommended to keep the size of the global context, the user information, and the feature flags below 3KiB. If the data exceeds this limit, a warning is displayed: `The data exceeds the recommended 3KiB threshold.`

Reviewer: ```suggestion
For the best user experience, Datadog recommends keeping the size of the global context, user information, and feature flags below 3KiB. If the data exceeds this limit, a warning is displayed: `The data exceeds the recommended 3KiB threshold.`
```
